### ISSUE TITLE
SuppressWarnings NullAway.Init not working as expected

### ISSUE THREAD

**Issue by fprochazka** [uber/NullAway#333]
I have a Hibernate entity (simplified)

```java
public class CustomerLicence
{

    @SuppressWarnings("NullAway.Init")
    private EnumSet<LicenceType> licenceTypes;

    public CustomerDataSourceLicence(
        final EnumSet<LicenceType> licenceTypes
    )
    {
        this.setLicenceTypes(licenceTypes);
    }

    @SuppressWarnings("NullAway.Init")
    protected CustomerDataSourceLicence()
    {
    }

    public void setLicenceTypes(final EnumSet<LicenceType> licenceTypes)
    {
        // complex validation
        this.licenceTypes = licenceTypes.clone();
    }
```

because of Hibernate's limitation, I have to have the `protected` constructor and also the class cannot be final, to allow auto-proxying. Therefore I've silenced the `protected` constructor - all fine so far.

The real problem is with the public constructor

```
[ERROR] CustomerDataSourceLicence.java:[49,12] [NullAway] initializer method does not guarantee @NonNull field licenceTypes is initialized along all control-flow paths (remember to check for exceptions or early returns).
```

I really want to keep the `setLicenceTypes` method, because I want to be able to change the value later and it has to be validated in the same way. I also understand that since the class and the method are not final, it's not guaranteed that the field will be set upon construction. So I've decided to annotate the field with `@SuppressWarnings("NullAway.Init")` (not the constructor, to keep the other fields still checked). But it doesn't seem to work, which is very confusing, since I've used the same field suppression in a different project and it worked nicely (context here: #313) 

<details>
 <summary>maven: effective pom</summary>

```xml
<plugin>
    <groupId>org.apache.maven.plugins</groupId>
    <artifactId>maven-compiler-plugin</artifactId>
    <version>3.8.1</version>
    <configuration combine.self="append">
        <compilerId>javac-with-errorprone</compilerId>
        <forceJavacCompilerUse>true</forceJavacCompilerUse>
        <source>1.8</source>
        <target>1.8</target>
        <compilerArgs combine.self="override">
            <arg>-Werror</arg>
            <arg>-Xlint:all,-fallthrough,-processing,-serial,-classfile</arg>
            <arg>-parameters</arg>
            <arg>-XDcompilePolicy=simple</arg>
            <arg>-XepDisableWarningsInGeneratedCode</arg>
            <arg>-XepOpt:Immutable:KnownImmutable=javax.money.CurrencyUnit</arg>
            <arg>-Xep:ConstantField:WARN</arg>
            <arg>-Xep:ClassName:WARN</arg>
            <arg>-Xep:DivZero:WARN</arg>
            <arg>-Xep:EmptyIf:WARN</arg>
            <arg>-Xep:IterablePathParameter:WARN</arg>
            <arg>-Xep:LongLiteralLowerCaseSuffix:WARN</arg>
            <arg>-Xep:AnnotationPosition:WARN</arg>
            <arg>-Xep:EmptyTopLevelDeclaration:WARN</arg>
            <arg>-Xep:EqualsBrokenForNull:WARN</arg>
            <arg>-Xep:FunctionalInterfaceClash:WARN</arg>
            <arg>-Xep:InvalidInlineTag:WARN</arg>
            <arg>-Xep:InvalidParam:WARN</arg>
            <arg>-Xep:MissingDefault:WARN</arg>
            <arg>-Xep:NonCanonicalStaticMemberImport:WARN</arg>
            <arg>-Xep:PrimitiveArrayPassedToVarargsMethod:WARN</arg>
            <arg>-Xep:RedundantOverride:WARN</arg>
            <arg>-Xep:RedundantThrows:WARN</arg>
            <arg>-Xep:StaticQualifiedUsingExpression:WARN</arg>
            <arg>-Xep:StringEquality:WARN</arg>
            <arg>-Xep:UnusedException:WARN</arg>
            <arg>-Xep:ConstantField:WARN</arg>
            <arg>-Xep:MultiVariableDeclaration:WARN</arg>
            <arg>-Xep:MultipleTopLevelClasses:WARN</arg>
            <arg>-Xep:MultipleUnaryOperatorsInMethodCall:WARN</arg>
            <arg>-Xep:PrivateConstructorForUtilityClass:WARN</arg>
            <arg>-Xep:UngroupedOverloads:WARN</arg>
            <arg>-Xep:NullAway:ERROR</arg>
            <arg>-XepOpt:NullAway:AnnotatedPackages=com.cogvio</arg>
            <arg>-XepOpt:NullAway:TreatGeneratedAsUnannotated=true</arg>
            <arg>-XepOpt:NullAway:AcknowledgeRestrictiveAnnotations=true</arg>
            <arg>-XepOpt:NullAway:KnownInitializers=org.springframework.beans.factory.InitializingBean.afterPropertiesSet</arg>
            <arg>-XepOpt:NullAway:ExcludedFieldAnnotations=javax.persistence.PersistenceContext,org.springframework.beans.factory.annotation.Autowired</arg>
        </compilerArgs>
        <annotationProcessorPaths combine.self="append">
            <path>
                <groupId>com.uber.nullaway</groupId>
                <artifactId>nullaway</artifactId>
                <version>0.7.5</version>
            </path>
        </annotationProcessorPaths>
        <showWarnings>true</showWarnings>
    </configuration>
    <dependencies>
        <dependencies>
          <dependency>
            <groupId>org.codehaus.plexus</groupId>
            <artifactId>plexus-compiler-javac-errorprone</artifactId>
            <version>2.8.5</version>
          </dependency>
          <dependency>
            <groupId>com.google.errorprone</groupId>
            <artifactId>error_prone_core</artifactId>
            <version>2.3.3</version>
          </dependency>
          <dependency>
            <!-- this only adds our custom LibraryModels -->
            <groupId>com.cogvio.pm</groupId>
            <artifactId>build-tools-nullaway</artifactId>
            <version>0.0.1-SNAPSHOT</version>
          </dependency>
        </dependencies>
    </dependencies>
</plugin>
```

</details>

**Comment by shas19** [comment]
I added a test, could not reproduce the problem. 

@fprochazka Am I missing anything? 

```
  public class SuppressWarningsD {

    @SuppressWarnings("NullAway.Init")
    private EnumSet<Object> f;

    SuppressWarningsD(final EnumSet<Object> f){
      this.setF(f);
    }

    @SuppressWarnings("NullAway.Init")
    protected SuppressWarningsD()
    {
    }

    public void setF(final EnumSet<Object> f)
    {
      // complex validation
      this.f = f.clone();
    }
  }
```
